site/bps/BP-43-gradle-migration.md

```diff
@@ -0,0 +1,63 @@
+---
+title: "BP-43: Gradle migration"
+issue: https://github.com/apache/bookkeeper/<issue-number>
+state: "Under Discussion"
+release: "N/A"
+---
+
+### Motivation
+- Current maven based build system is very slow. It takes about 4m30 to execute
+   mvn clean package -DskipTests while a gradle counterpart 
+  `./gradlew clean jar` completes in 40 seconds on the same hardware. Using the gradle build cache.
+- Incremental build- Gradle provides better incremental build support. Subsequent builds run in very little time.
+- Caching of tests execution - Gradle build cache also keeps track of test successes with respect to the dependencies and can do test retries of just what has failed.
+- Better handling of multi module projects - Bookkeeper is a multi module project where bookkeeper-server module for an example depends on following modules 
+   - bookkeeper-stats
+   - cpu-affinity
+   - bookkeeper-common
+   - bookkeeper-common-allocator
+   - bookkeeper-http:http-server
+   - bookkeeper-proto
+   - circe-checksum
+   - bookkeeper-tools-framework
+   
+  If change is made on any of the above mentioned dependent modules, building module `bookkeeper-server` will pick up those changes. 
+  While in case of maven changed modules had to be built explicitly in order for the changes to be picked up.
+  For instance If a change is made in `bookkeeper-common` with gradle 
+  `./gradlew bookkeeper-server:test` will pick up those changes and run tests based on those changes. 
+  While on the other hands for maven `mvn  install bookkeeper-common` had to run before running mvn build `bookkeeper-server`
+
+_Describe the problems you are trying to solve_
+
+### Public Interfaces
+N/A
+### Proposed Changes
+- Every module and sub module which are currently being built using maven would be built using gradle.
+- Run all the unit tests, functional tests etc using gradle.
+- Integrating gradle build to CI/CD pipeline 
+- Build ASF release capability using gradle
+- Remove Maven build
+
+### Compatibility, Deprecation, and Migration Plan
```

Comment:
documentation update should be a part of the plan (contribution guide/IDE setup, release guide)

Reply:
Sure. I will add that.

Reply:
Added.